go/toolchains.rst

```diff
@@ -225,6 +235,10 @@ SDK version to use (for example, :value:`"1.15.5"`).
 | Normally this is set to a Go version like :value:`"1.15.5"`. It may also be                      |
 | set to :value:`"host"`, which will cause rules_go to use the Go toolchain                        |
 | installed on the host system (found using ``GOROOT`` or ``PATH``).                               |
+|                                                                                                  |
+| If ``version`` is specified and is not set to :value:`"host"`, the SDK will be lazily fetched    |
```

Comment:
Sorry for not seeing this earlier, but I think we can clarify this a bit more. How about:

"If version ..., the SDK will only be fetched when the build uses a Go toolchain and [toolchain resolution](https://bazel.build/extending/toolchains#toolchain-resolution) results in this SDK being chosen. Otherwise, it will be fetched unconditionally.

I could see "lazily" and "eagerly" being slightly confusing here as "lazily" can very well mean "never" and "eagerly" can mean "always". I would also like to avoid referring to the "analysis phase" if we don't have to.